### PR TITLE
Search past online events

### PR DESCRIPTION
**Pull request by skumarqa2010**
Search past online events

### CODE REVIEW

File: cypress/integration/event-signup-flow.spec.js [hunk 1]
```diff
@@ -747,4 +747,23 @@ describe("Verify page load " + new Date(), () => {
...
+	});
+	it("It shows past events only", function () {
+		cy.navigateToPage("/events/category/online-events/archive");
+		cy.getPastEventDateAndTime().each((eventDateAndTime, index, $list) => {
+			eventDateAndTime = eventDateAndTime.text().split(":")[0];
+			eventDateAndTime = eventDateAndTime.substring(0, eventDateAndTime.lastIndexOf(" "));
+			var eventYear = new Date(eventDateAndTime).getFullYear();
+			var eventMonth = new Date(eventDateAndTime).getMonth();
+			var eventDate = new Date(eventDateAndTime).getDate();
+			cy.compareEventDate(eventYear, eventMonth, eventDate);
```
Comment (ethax-ross) [inline cypress/integration/event-signup-flow.spec.js:766]: Not sure what this is doing? It looks like it gets the first event date/time and makes sure it's in the past? Would it be better to extract all the date/times from events, loop them and check they are all in the past?

Comment (skumarqa2010) [reply cypress/integration/event-signup-flow.spec.js]: Resolved

File: cypress/integration/event-signup-flow.spec.js [hunk 2]
```diff
@@ -747,4 +747,26 @@ describe("Verify page load " + new Date(), () => {
...
+			.should("have.text", "Search for Past Online Events");
+	});
+	it("It shows past events only", function () {
+		cy.navigateToPage("/events/category/online-events/archive");
+		var today = new Date();
+		var currentDate = today.getDate() + " " + today.getMonth() + " " + today.getFullYear();
+		cy.getPastEventDateAndTime().each((eventDateAndTime, index, $list) => {
+			eventDateAndTime = eventDateAndTime.text().split(":")[0];
+			var eventDate = eventDateAndTime.substring(0, eventDateAndTime.lastIndexOf(" "));
+			if (eventDate < currentDate) {
```
Comment (ethax-ross) [inline cypress/integration/event-signup-flow.spec.js:765]: Can `getPastEventDateAndTime` parse the date text into a `Date` object and then we can simplify this quite a bit to be:

```
expect(eventDate.getTime()).toBeLessThan(new Date().getTime())
```

Comment (skumarqa2010) [reply cypress/integration/event-signup-flow.spec.js]: Fixed.

**Review by ethax-ross — APPROVED** [review]

**Pull request merged**